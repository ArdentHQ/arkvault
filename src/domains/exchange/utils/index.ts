export * from "./assertions";
export * from "./errors";
