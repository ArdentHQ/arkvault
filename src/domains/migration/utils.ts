import { MigrationTransactionStatus } from "@/domains/migration/migration.contracts";

export const getIcon = (status: MigrationTransactionStatus) => {
	if (status === MigrationTransactionStatus.Confirmed) {
		return {
			color: "text-theme-success-600",
			name: "CircleCheckMark",
		};
	}

	return {
		color: "text-theme-warning-300",
		name: "Clock",
	};
};
