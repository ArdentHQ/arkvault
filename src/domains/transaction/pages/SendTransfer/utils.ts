export const getFeeType = (numberOfRecipients: number): "multiPayment" | "transfer" =>
	numberOfRecipients > 1 ? "multiPayment" : "transfer";
