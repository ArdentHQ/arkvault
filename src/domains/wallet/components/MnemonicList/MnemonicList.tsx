import React, { useMemo } from "react";

import { Skeleton } from "@/app/components/Skeleton";

interface MnemonicListProperties {
	mnemonic: string;
}

export const MnemonicList: React.VFC<MnemonicListProperties> = ({ mnemonic }) => {
	let mnemonicWords: string[];

	// Check for Japanese "space"
	mnemonicWords = /\u3000/.test(mnemonic) ? mnemonic.split("\u3000") : mnemonic.split(" ");

	return (
		<ul className="grid grid-cols-2 gap-x-3 gap-y-5 sm:grid-cols-4">
			{mnemonicWords.map((word, index) => (
				<li
					data-testid="MnemonicList__item"
					key={index}
					className="relative flex items-center rounded border border-theme-secondary-400 p-2 dark:border-theme-secondary-700 sm:p-4"
				>
					<span className="absolute left-0 top-0 hidden -translate-y-2 translate-x-2 bg-theme-background px-1 text-xs text-theme-secondary-700 sm:block">
						{index + 1}
					</span>
					<div className="ml-1 mr-4 block text-xs text-theme-secondary-700 sm:hidden">{index + 1}</div>
					<div className="sm:text-md text-sm">{word}</div>
				</li>
			))}
		</ul>
	);
};

export const MnemonicListSkeleton: React.VFC = () => {
	const skeletons = useMemo(
		() =>
			Array.from({ length: 24 }).map(() => {
				const [min, max] = [50, 70];

				return Math.floor(Math.random() * (max - min + 1) + min);
			}),
		[],
	);

	return (
		<ul className="grid grid-cols-2 gap-x-3 gap-y-5 sm:grid-cols-4">
			{skeletons.map((width, index) => (
				<li
					data-testid="MnemonicList__item_skeleton"
					key={index}
					className="relative flex items-center rounded border border-theme-secondary-300 p-2 dark:border-theme-secondary-700 sm:p-4"
				>
					<span className="absolute left-0 top-0 hidden -translate-y-2 translate-x-2 bg-theme-background px-1 text-xs text-theme-secondary-700 sm:block">
						{index + 1}
					</span>
					<div className="ml-1 mr-4 block text-xs text-theme-secondary-700 sm:hidden">{index + 1}</div>
					<div>
						<Skeleton width={width} height={20} />
					</div>
				</li>
			))}
		</ul>
	);
};
