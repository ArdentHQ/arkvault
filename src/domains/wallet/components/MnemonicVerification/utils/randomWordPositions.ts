import { sample } from "@ardenthq/sdk-helpers";

const randomWordPositions = (length: number): number[] => {
	const positions: number[] = [...Array.from({ length }).keys()];
	const result: number[] = [];

	while (result.length < 3) {
		const randomNumber = sample(positions) + 1;

		if (result.includes(randomNumber)) {
			continue;
		}

		result.push(randomNumber);
	}

	return result.sort((a, b) => a - b);
};

export { randomWordPositions };
