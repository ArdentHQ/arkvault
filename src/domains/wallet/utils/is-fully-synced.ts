import { Contracts } from "@payvo/sdk-profiles";

export const isFullySynced = (wallet: Contracts.IReadWriteWallet): boolean =>
	wallet.hasBeenFullyRestored() && (wallet.hasSyncedWithNetwork() || wallet.balance() === 0);
