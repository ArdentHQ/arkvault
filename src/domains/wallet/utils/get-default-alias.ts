import { Networks } from "@ardenthq/sdk";
import { Contracts } from "@ardenthq/sdk-profiles";
import { networkDisplayName } from "@/utils/network-utils";

interface GetDefaultAliasInput {
	profile: Contracts.IProfile;
	network: Networks.Network;
}

export const getDefaultAlias = ({ profile, network }: GetDefaultAliasInput): string => {
	const makeAlias = (count: number) => `${networkDisplayName(network)} #${count}`;

	const sameCoinWallets = profile.wallets().findByCoinWithNetwork(network.coin(), network.id());

	let counter = sameCoinWallets.length;

	if (counter === 0) {
		counter = 1;
	}

	while (profile.wallets().findByAlias(makeAlias(counter))) {
		counter++;
	}

	return makeAlias(counter);
};
