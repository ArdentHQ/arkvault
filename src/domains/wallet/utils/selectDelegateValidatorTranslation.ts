import { Networks } from "@ardenthq/sdk";
import { isMainsailNetwork } from "@/utils/network-utils";

interface Properties<T extends string | React.ReactNode> {
	network: Networks.Network;
	validatorStr: T;
	delegateStr: T;
}

export const selectDelegateValidatorTranslation = <T extends string | React.ReactNode>({ network, validatorStr, delegateStr }: Properties<T>): T => isMainsailNetwork(network) ? validatorStr : delegateStr;



