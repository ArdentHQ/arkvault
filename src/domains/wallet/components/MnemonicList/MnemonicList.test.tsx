import React from "react";

import { MnemonicList } from "./MnemonicList";
import { render, screen } from "@/utils/testing-library";

describe("MnemonicList", () => {
	it("should render", () => {
		const mnemonic = "stomach practice violin later term trend suit switch inch theory hundred system";
		const words = mnemonic.split(" ");

		const { asFragment } = render(<MnemonicList mnemonic={mnemonic} />);

		const items = screen.getAllByTestId("MnemonicList__item");

		expect(items).toHaveLength(words.length);

		for (const [index, item] of items.entries()) {
			expect(item).toHaveTextContent(words[index]);
		}

		expect(asFragment()).toMatchSnapshot();
	});

	it("should render with special delimiter", () => {
		const mnemonic =
			"あさい　せつでん　さいかい　にんち　たんけん　ぬまえび　こうじ　こっか　はあく　げきか　ふめつ　ちらし";
		const words = mnemonic.split("\u3000");

		const { asFragment } = render(<MnemonicList mnemonic={mnemonic} />);

		const items = screen.getAllByTestId("MnemonicList__item");

		expect(items).toHaveLength(words.length);

		for (const [index, item] of items.entries()) {
			expect(item).toHaveTextContent(words[index]);
		}

		expect(asFragment()).toMatchSnapshot();
	});
});
