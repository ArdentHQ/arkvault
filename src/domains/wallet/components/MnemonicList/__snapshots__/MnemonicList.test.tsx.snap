// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MnemonicList > should render 1`] = `
<DocumentFragment>
  <ul
    class="grid grid-cols-2 gap-x-3 gap-y-5 sm:grid-cols-4"
  >
    <li
      class="relative flex items-center rounded border border-theme-secondary-400 p-2 dark:border-theme-secondary-700 sm:p-4"
      data-testid="MnemonicList__item"
    >
      <span
        class="absolute top-0 left-0 hidden translate-x-2 -translate-y-2 bg-theme-background px-1 text-xs text-theme-secondary-700 sm:block"
      >
        1
      </span>
      <div
        class="ml-1 mr-4 block text-xs text-theme-secondary-700 sm:hidden"
      >
        1
      </div>
      <div
        class="sm:text-md text-sm"
      >
        stomach
      </div>
    </li>
    <li
      class="relative flex items-center rounded border border-theme-secondary-400 p-2 dark:border-theme-secondary-700 sm:p-4"
      data-testid="MnemonicList__item"
    >
      <span
        class="absolute top-0 left-0 hidden translate-x-2 -translate-y-2 bg-theme-background px-1 text-xs text-theme-secondary-700 sm:block"
      >
        2
      </span>
      <div
        class="ml-1 mr-4 block text-xs text-theme-secondary-700 sm:hidden"
      >
        2
      </div>
      <div
        class="sm:text-md text-sm"
      >
        practice
      </div>
    </li>
    <li
      class="relative flex items-center rounded border border-theme-secondary-400 p-2 dark:border-theme-secondary-700 sm:p-4"
      data-testid="MnemonicList__item"
    >
      <span
        class="absolute top-0 left-0 hidden translate-x-2 -translate-y-2 bg-theme-background px-1 text-xs text-theme-secondary-700 sm:block"
      >
        3
      </span>
      <div
        class="ml-1 mr-4 block text-xs text-theme-secondary-700 sm:hidden"
      >
        3
      </div>
      <div
        class="sm:text-md text-sm"
      >
        violin
      </div>
    </li>
    <li
      class="relative flex items-center rounded border border-theme-secondary-400 p-2 dark:border-theme-secondary-700 sm:p-4"
      data-testid="MnemonicList__item"
    >
      <span
        class="absolute top-0 left-0 hidden translate-x-2 -translate-y-2 bg-theme-background px-1 text-xs text-theme-secondary-700 sm:block"
      >
        4
      </span>
      <div
        class="ml-1 mr-4 block text-xs text-theme-secondary-700 sm:hidden"
      >
        4
      </div>
      <div
        class="sm:text-md text-sm"
      >
        later
      </div>
    </li>
    <li
      class="relative flex items-center rounded border border-theme-secondary-400 p-2 dark:border-theme-secondary-700 sm:p-4"
      data-testid="MnemonicList__item"
    >
      <span
        class="absolute top-0 left-0 hidden translate-x-2 -translate-y-2 bg-theme-background px-1 text-xs text-theme-secondary-700 sm:block"
      >
        5
      </span>
      <div
        class="ml-1 mr-4 block text-xs text-theme-secondary-700 sm:hidden"
      >
        5
      </div>
      <div
        class="sm:text-md text-sm"
      >
        term
      </div>
    </li>
    <li
      class="relative flex items-center rounded border border-theme-secondary-400 p-2 dark:border-theme-secondary-700 sm:p-4"
      data-testid="MnemonicList__item"
    >
      <span
        class="absolute top-0 left-0 hidden translate-x-2 -translate-y-2 bg-theme-background px-1 text-xs text-theme-secondary-700 sm:block"
      >
        6
      </span>
      <div
        class="ml-1 mr-4 block text-xs text-theme-secondary-700 sm:hidden"
      >
        6
      </div>
      <div
        class="sm:text-md text-sm"
      >
        trend
      </div>
    </li>
    <li
      class="relative flex items-center rounded border border-theme-secondary-400 p-2 dark:border-theme-secondary-700 sm:p-4"
      data-testid="MnemonicList__item"
    >
      <span
        class="absolute top-0 left-0 hidden translate-x-2 -translate-y-2 bg-theme-background px-1 text-xs text-theme-secondary-700 sm:block"
      >
        7
      </span>
      <div
        class="ml-1 mr-4 block text-xs text-theme-secondary-700 sm:hidden"
      >
        7
      </div>
      <div
        class="sm:text-md text-sm"
      >
        suit
      </div>
    </li>
    <li
      class="relative flex items-center rounded border border-theme-secondary-400 p-2 dark:border-theme-secondary-700 sm:p-4"
      data-testid="MnemonicList__item"
    >
      <span
        class="absolute top-0 left-0 hidden translate-x-2 -translate-y-2 bg-theme-background px-1 text-xs text-theme-secondary-700 sm:block"
      >
        8
      </span>
      <div
        class="ml-1 mr-4 block text-xs text-theme-secondary-700 sm:hidden"
      >
        8
      </div>
      <div
        class="sm:text-md text-sm"
      >
        switch
      </div>
    </li>
    <li
      class="relative flex items-center rounded border border-theme-secondary-400 p-2 dark:border-theme-secondary-700 sm:p-4"
      data-testid="MnemonicList__item"
    >
      <span
        class="absolute top-0 left-0 hidden translate-x-2 -translate-y-2 bg-theme-background px-1 text-xs text-theme-secondary-700 sm:block"
      >
        9
      </span>
      <div
        class="ml-1 mr-4 block text-xs text-theme-secondary-700 sm:hidden"
      >
        9
      </div>
      <div
        class="sm:text-md text-sm"
      >
        inch
      </div>
    </li>
    <li
      class="relative flex items-center rounded border border-theme-secondary-400 p-2 dark:border-theme-secondary-700 sm:p-4"
      data-testid="MnemonicList__item"
    >
      <span
        class="absolute top-0 left-0 hidden translate-x-2 -translate-y-2 bg-theme-background px-1 text-xs text-theme-secondary-700 sm:block"
      >
        10
      </span>
      <div
        class="ml-1 mr-4 block text-xs text-theme-secondary-700 sm:hidden"
      >
        10
      </div>
      <div
        class="sm:text-md text-sm"
      >
        theory
      </div>
    </li>
    <li
      class="relative flex items-center rounded border border-theme-secondary-400 p-2 dark:border-theme-secondary-700 sm:p-4"
      data-testid="MnemonicList__item"
    >
      <span
        class="absolute top-0 left-0 hidden translate-x-2 -translate-y-2 bg-theme-background px-1 text-xs text-theme-secondary-700 sm:block"
      >
        11
      </span>
      <div
        class="ml-1 mr-4 block text-xs text-theme-secondary-700 sm:hidden"
      >
        11
      </div>
      <div
        class="sm:text-md text-sm"
      >
        hundred
      </div>
    </li>
    <li
      class="relative flex items-center rounded border border-theme-secondary-400 p-2 dark:border-theme-secondary-700 sm:p-4"
      data-testid="MnemonicList__item"
    >
      <span
        class="absolute top-0 left-0 hidden translate-x-2 -translate-y-2 bg-theme-background px-1 text-xs text-theme-secondary-700 sm:block"
      >
        12
      </span>
      <div
        class="ml-1 mr-4 block text-xs text-theme-secondary-700 sm:hidden"
      >
        12
      </div>
      <div
        class="sm:text-md text-sm"
      >
        system
      </div>
    </li>
  </ul>
</DocumentFragment>
`;

exports[`MnemonicList > should render with special delimiter 1`] = `
<DocumentFragment>
  <ul
    class="grid grid-cols-2 gap-x-3 gap-y-5 sm:grid-cols-4"
  >
    <li
      class="relative flex items-center rounded border border-theme-secondary-400 p-2 dark:border-theme-secondary-700 sm:p-4"
      data-testid="MnemonicList__item"
    >
      <span
        class="absolute top-0 left-0 hidden translate-x-2 -translate-y-2 bg-theme-background px-1 text-xs text-theme-secondary-700 sm:block"
      >
        1
      </span>
      <div
        class="ml-1 mr-4 block text-xs text-theme-secondary-700 sm:hidden"
      >
        1
      </div>
      <div
        class="sm:text-md text-sm"
      >
        あさい
      </div>
    </li>
    <li
      class="relative flex items-center rounded border border-theme-secondary-400 p-2 dark:border-theme-secondary-700 sm:p-4"
      data-testid="MnemonicList__item"
    >
      <span
        class="absolute top-0 left-0 hidden translate-x-2 -translate-y-2 bg-theme-background px-1 text-xs text-theme-secondary-700 sm:block"
      >
        2
      </span>
      <div
        class="ml-1 mr-4 block text-xs text-theme-secondary-700 sm:hidden"
      >
        2
      </div>
      <div
        class="sm:text-md text-sm"
      >
        せつでん
      </div>
    </li>
    <li
      class="relative flex items-center rounded border border-theme-secondary-400 p-2 dark:border-theme-secondary-700 sm:p-4"
      data-testid="MnemonicList__item"
    >
      <span
        class="absolute top-0 left-0 hidden translate-x-2 -translate-y-2 bg-theme-background px-1 text-xs text-theme-secondary-700 sm:block"
      >
        3
      </span>
      <div
        class="ml-1 mr-4 block text-xs text-theme-secondary-700 sm:hidden"
      >
        3
      </div>
      <div
        class="sm:text-md text-sm"
      >
        さいかい
      </div>
    </li>
    <li
      class="relative flex items-center rounded border border-theme-secondary-400 p-2 dark:border-theme-secondary-700 sm:p-4"
      data-testid="MnemonicList__item"
    >
      <span
        class="absolute top-0 left-0 hidden translate-x-2 -translate-y-2 bg-theme-background px-1 text-xs text-theme-secondary-700 sm:block"
      >
        4
      </span>
      <div
        class="ml-1 mr-4 block text-xs text-theme-secondary-700 sm:hidden"
      >
        4
      </div>
      <div
        class="sm:text-md text-sm"
      >
        にんち
      </div>
    </li>
    <li
      class="relative flex items-center rounded border border-theme-secondary-400 p-2 dark:border-theme-secondary-700 sm:p-4"
      data-testid="MnemonicList__item"
    >
      <span
        class="absolute top-0 left-0 hidden translate-x-2 -translate-y-2 bg-theme-background px-1 text-xs text-theme-secondary-700 sm:block"
      >
        5
      </span>
      <div
        class="ml-1 mr-4 block text-xs text-theme-secondary-700 sm:hidden"
      >
        5
      </div>
      <div
        class="sm:text-md text-sm"
      >
        たんけん
      </div>
    </li>
    <li
      class="relative flex items-center rounded border border-theme-secondary-400 p-2 dark:border-theme-secondary-700 sm:p-4"
      data-testid="MnemonicList__item"
    >
      <span
        class="absolute top-0 left-0 hidden translate-x-2 -translate-y-2 bg-theme-background px-1 text-xs text-theme-secondary-700 sm:block"
      >
        6
      </span>
      <div
        class="ml-1 mr-4 block text-xs text-theme-secondary-700 sm:hidden"
      >
        6
      </div>
      <div
        class="sm:text-md text-sm"
      >
        ぬまえび
      </div>
    </li>
    <li
      class="relative flex items-center rounded border border-theme-secondary-400 p-2 dark:border-theme-secondary-700 sm:p-4"
      data-testid="MnemonicList__item"
    >
      <span
        class="absolute top-0 left-0 hidden translate-x-2 -translate-y-2 bg-theme-background px-1 text-xs text-theme-secondary-700 sm:block"
      >
        7
      </span>
      <div
        class="ml-1 mr-4 block text-xs text-theme-secondary-700 sm:hidden"
      >
        7
      </div>
      <div
        class="sm:text-md text-sm"
      >
        こうじ
      </div>
    </li>
    <li
      class="relative flex items-center rounded border border-theme-secondary-400 p-2 dark:border-theme-secondary-700 sm:p-4"
      data-testid="MnemonicList__item"
    >
      <span
        class="absolute top-0 left-0 hidden translate-x-2 -translate-y-2 bg-theme-background px-1 text-xs text-theme-secondary-700 sm:block"
      >
        8
      </span>
      <div
        class="ml-1 mr-4 block text-xs text-theme-secondary-700 sm:hidden"
      >
        8
      </div>
      <div
        class="sm:text-md text-sm"
      >
        こっか
      </div>
    </li>
    <li
      class="relative flex items-center rounded border border-theme-secondary-400 p-2 dark:border-theme-secondary-700 sm:p-4"
      data-testid="MnemonicList__item"
    >
      <span
        class="absolute top-0 left-0 hidden translate-x-2 -translate-y-2 bg-theme-background px-1 text-xs text-theme-secondary-700 sm:block"
      >
        9
      </span>
      <div
        class="ml-1 mr-4 block text-xs text-theme-secondary-700 sm:hidden"
      >
        9
      </div>
      <div
        class="sm:text-md text-sm"
      >
        はあく
      </div>
    </li>
    <li
      class="relative flex items-center rounded border border-theme-secondary-400 p-2 dark:border-theme-secondary-700 sm:p-4"
      data-testid="MnemonicList__item"
    >
      <span
        class="absolute top-0 left-0 hidden translate-x-2 -translate-y-2 bg-theme-background px-1 text-xs text-theme-secondary-700 sm:block"
      >
        10
      </span>
      <div
        class="ml-1 mr-4 block text-xs text-theme-secondary-700 sm:hidden"
      >
        10
      </div>
      <div
        class="sm:text-md text-sm"
      >
        げきか
      </div>
    </li>
    <li
      class="relative flex items-center rounded border border-theme-secondary-400 p-2 dark:border-theme-secondary-700 sm:p-4"
      data-testid="MnemonicList__item"
    >
      <span
        class="absolute top-0 left-0 hidden translate-x-2 -translate-y-2 bg-theme-background px-1 text-xs text-theme-secondary-700 sm:block"
      >
        11
      </span>
      <div
        class="ml-1 mr-4 block text-xs text-theme-secondary-700 sm:hidden"
      >
        11
      </div>
      <div
        class="sm:text-md text-sm"
      >
        ふめつ
      </div>
    </li>
    <li
      class="relative flex items-center rounded border border-theme-secondary-400 p-2 dark:border-theme-secondary-700 sm:p-4"
      data-testid="MnemonicList__item"
    >
      <span
        class="absolute top-0 left-0 hidden translate-x-2 -translate-y-2 bg-theme-background px-1 text-xs text-theme-secondary-700 sm:block"
      >
        12
      </span>
      <div
        class="ml-1 mr-4 block text-xs text-theme-secondary-700 sm:hidden"
      >
        12
      </div>
      <div
        class="sm:text-md text-sm"
      >
        ちらし
      </div>
    </li>
  </ul>
</DocumentFragment>
`;
